# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsRange.h
# Commit: 5480477f3599
# Full Hash: 5480477f35993b93846c693c8e388f727ea18dc4
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2024-06-21 04:13:05
# Regressor Bug: 1881096
# File Overlap Count: 2
# Description:
#   Bug 1881096 - Add basic support of character removal to CrossShadowBoundaryRange r=emilio,jjaschke,dom-core
#   
#   Because nsPrintJob would remove characters from the selected boundary
#   nodes, the offsets of the boundaries needs to be updated if this
#   happens.
# ==============================================================================

diff -r 2d732ba0d3b6 -r 5480477f3599 dom/base/nsRange.h
--- a/dom/base/nsRange.h	Thu Jun 20 20:15:50 2024 +0000
+++ b/dom/base/nsRange.h	Thu Jun 20 20:15:51 2024 +0000
@@ -297,6 +297,11 @@
                                JS::Handle<JSObject*> aGivenProto) final;
   DocGroup* GetDocGroup() const;
 
+  // Given a CharacterDataChangeInfo and an RangeBoundary of where the
+  // character changes occurred at, compute the new boundary.
+  static RawRangeBoundary ComputeNewBoundaryWhenBoundaryInsideChangedText(
+      const CharacterDataChangeInfo& aInfo, const RawRangeBoundary& aBoundary);
+
  private:
   // no copy's or assigns
   nsRange(const nsRange&);
